js2-mode.el

```diff
@@ -8130,11 +8137,23 @@ Scanner should be initialized."
     (setq js2-mode-ast root)  ; Make sure this is available for callbacks.
     ;; Give extensions a chance to muck with things before highlighting starts.
     (let ((js2-additional-externs js2-additional-externs))
+      (js2-filter-parsed-errors)
       (save-excursion
         (run-hooks 'js2-post-parse-callbacks))
       (js2-highlight-undeclared-vars))
     root))
 
+(defun js2-filter-parsed-errors ()
+  "Remove `js2-ignored-warnings' from `js2-parsed-errors'."
```

Comment:
Better:

Remove `js2-parsed-warnings` elements that match `js2-ignored-warnings`.